File: hash_1.py
# 완주하지 못한 선수 찾기

def solution(participant, completion):
  d = {}
  for x in participant:
    d[x] = d.get(x, 0) + 1
  for x in completion:
    d[x] -= 1
  dnf = [k for k, v in d.items() if v > 0]
  answer = dnf[0]
  return answer

participant = ["jace", "jonus"]
completion = ["jonus"]

print(solution(participant, completion))

File: src/Hash1.java
import java.util.HashMap;

public class Hash1 {
    public static String solution(String[] participant, String[] completion) {
        HashMap map = new HashMap<String, Integer>();
        for (String val : participant) {
            if (map.containsKey(val)) {
                map.put(val, (int)map.get(val) + 1);
            } else {
                map.put(val, 1);
            }
        }

        for (String val : completion) {
            map.put(val, (int)map.get(val) - 1);
        }

        String[] answer = {""};
        map.forEach((key, value) -> {
            if ((int)value == 1) {
                answer[0] = (String) key;
            }
        });
        return answer[0];
    }

    public static void main(String[] args) {
        String[] participant = {"marina", "josipa", "nikola", "vinko", "filipa"};
        String[] completion = {"josipa", "filipa", "marina", "nikola"};
        System.out.println(solution(participant, completion));
    }
}
